tests/python_package_test/utils.py

```diff
@@ -0,0 +1,17 @@
+try:
+    from functools import lru_cache
+except ImportError:
+    import warnings
+    warnings.warn("Could not import functools.lru_cache", RuntimeWarning)
+
+    def lru_cache(maxsize=None):
```

Comment:
Please memoize this too
https://github.com/microsoft/LightGBM/blob/5cc9e671977f67126efe4196d6161ed5bd40aaca/tests/python_package_test/test_plotting.py#L19

Reply:
what would be the purpose? That's the only call to `load_breast_cancer()` in that module. So I think the caching would add a tiny bit of overhead for no benefit.

Reply:
How is it differs with other calls you've memoized? 5 calls of `load_breast_cancer()` in `test_plotting.py` is even more than 3 calls of the same function in `test_basic.py`, for example.

Reply:
there is only 1 call in `test_plotting.py`

```shell
git grep load_breast_cancer tests/python_package_test/test_plotting.py
```

![image](https://user-images.githubusercontent.com/7608904/97510753-699fea80-1953-11eb-994e-18e7c7c9d9d6.png)

Reply:
Yes, you are right, but the method in which this call is performed (`setUp`) is called before each test. So, actually we have 5 calls.

Reply:
OOOOOOOOOOO haha ok. I haven't used `unittest.TestCase` in a while, I forgot which one was a  "run before every test" setup and which one was a "run exactly once, before any tests" one.

Ok yes I'll update this

Reply:
added in https://github.com/microsoft/LightGBM/pull/3486/commits/dfb0fd35cdea555d8b8aac800fbb01528fbe0ca4

Reply:
Thanks! Actually I think we should refactor this to "run exactly once, before any tests" (`setUpClass()`), but it is another issue, of course.